app/models/grda_warehouse/auth_policies/source_client_policy.rb

```diff
@@ -83,7 +85,19 @@ def add_legacy_data_source_permissions(results)
     # is the client in a window data source?
     return unless context.legacy_window_data_source_ids.include?(client.data_source_id)
 
-    # check roi if the window config requires release (the "can_*_with_roi" permissions are not relevant in this case)
+    # The 'can_view_clients' and 'can_search_all_clients' permissions allow the user's permissions to apply to clients in windowed data sources, regardless of ROI.
+    # For example a user with can_view_clients and can_view_name can view the names of all clients in windowed data sources
+    if legacy_permissions.include?(:can_view_clients)
+      # all the legacy perms apply to the client
+      results.merge(legacy_permissions)
+      # early return since there's no point in checking ROI
+      return
+    elsif legacy_permissions.include?(:can_search_all_clients)
+      # can_search_all_clients confers a reduced set or permissions
```

Comment:
Historically this hasn't conferred fewer permissions, maybe we should make a note here that this is tighter than the historic permissions, but that's ok, since the search results have limited client details.  This is to enforce more strict permissions in the future.